src/plone/restapi/profiles/default/rolemap.xml

```diff
@@ -9,6 +9,11 @@
     </permission>
     <permission name="plone.restapi: Access Plone user information" acquire="True">
       <role name="Manager"/>
+      <role name="Site Administrator"/>
+    </permission>
+    <permission name="Manage users" acquire="True">
+      <role name="Manager"/>
+      <role name="Site Administrator"/>
```

Comment:
@wesleybl I see that in the rest of the PR, you've mostly changed the endpoints to check plone.app.controlpanel.UsersAndGroups rather than "Manage users". Is there something that "Manage users" is still needed for?

If so we'll need to take a closer look at how to solve it, because I don't think we can safely give the "Manage users" permission to Site Administrators. If I'm not mistaken, it provides access to some methods outside of our control (i.e. in PAS and the base Zope user folder) which could be used to escalate privileges by adding the Manager role to oneself.

If you can point me to where it's still preventing a Site Admin from doing something with the rest of your changes in place, I can try to give advice on how to fix it.

(Side note: if adding this permission were the correct thing to do, it should happen in CMFPlone rather than here. It's too confusing if a role has different permissions depending on whether or not plone.restapi is installed.)

Reply:
@wesleybl Ah, I found your comment about this here: https://github.com/plone/volto/issues/3917#issuecomment-1721836345

At first glance, I think we should resolve that by making plone.restapi also use the deleteMembers function from CMFPlone which avoids the "Manage users" check in CMFCore. The function even has a comment to remind us that that's what it is there for: https://github.com/plone/Products.CMFPlone/blob/bad7ee61597c72f165f5db979ee04efd06066d22/Products/CMFPlone/controlpanel/browser/usergroups_usersoverview.py#L225-L227

Reply:
@davisagli the delete users service allows you not to delete [memberareas](https://github.com/plone/plone.restapi/blob/48d2569b00b4e166556b6699ce4ee086f30a2c7e/src/plone/restapi/services/users/delete.py#L34-L36) and [localroles](https://github.com/plone/plone.restapi/blob/48d2569b00b4e166556b6699ce4ee086f30a2c7e/src/plone/restapi/services/users/delete.py#L38-L40). `deleteMembers` [always does this](https://github.com/plone/Products.CMFPlone/blob/bad7ee61597c72f165f5db979ee04efd06066d22/Products/CMFPlone/controlpanel/browser/usergroups_usersoverview.py#L252C33-L261). Should we keep this option in the service? If so, you will not be able to use `deleteMembers` directly.

Reply:
@wesleybl We should keep those options. We can copy a portion of the code from `deleteMembers` into the plone.restapi user delete service. It's a small amount of code duplication but probably worth it to avoid needing to deal with the complexity of making changes to `deleteMembers` in multiple branches of CMFPlone.

Reply:
@davisagli I kept it. Please take a look at https://github.com/plone/plone.restapi/pull/1712/commits/51eab0caef415490b46839f45b97fa680e42f972